DOCS-8072-4.3-tuning-backend-server-kt (#1412)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/tuning-backend-server.adoc

> = Backend Server Response Time

Reviewer: ```suggestion
= How Backend Server Response Time Affects Application Performance
```
I wonder if this topic belongs in one of the others.
It seems like a prerequisite to the whole big issue of application performance, to be considered before reading or doing anything with the app itself.

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> When investigating performance issues, you should first look if the overall flow's performance is heavily constrained by the latency (response time or message round-trip time in milliseconds unless otherwise noted) of the remote hosts to which the flow connects into.

Reviewer: The following two sentences summarize all the rest:
```suggestion
To troubleshoot performance issues, first determine whether flow performance is heavily constrained by the latency of the remote hosts to which the flow connects. Failing to do this before attempting to optimize your application results in wasted time and negligible performance improvement.
```
Your audience should already understand "latency" (if I understand it, everyone else does!).

Reviewer: I suggest omitting all the rest of this in favor of the rewording suggested above.
But I have copyedited the existing text in case you are uncomfortable rewording right now (but please save in a JIRA for future consideration).

> Failing to acknowledge the performance overhead that is added to a flow by the underlying complexities of the communication between the application and a remote host, may lead the optimization effort through the wrong path with negligible improvements in the end. +

Reviewer: SUGGEST REPLACING, but...:
```suggestion
Failure to acknowledge the performance overhead added to a flow by the underlying complexities of the communication between the application and a remote host might mislead your optimization efforts and result in negligible improvement. +
```

> Therefore, it is important that before starting any optimization in the Mule app, to first characterize the backend servers average latency and throughput, and to establish if these are the ones limiting the scalability or performance of your application.

Reviewer: SUGGEST REPLACING because it is repetitive, but:
```suggestion
Therefore, it is important that, before you begin any optimization in the Mule app, you first characterize the backend servers' average latency and throughput to determine if this is what is limiting your application's scalability or performance.
```

> There are many factors to consider such as: +

Reviewer: ```suggestion
Some of the factors to consider are: +
```

> * Average payload size exchanged between the app and the remote host.

Reviewer: ```suggestion
* Average size of the payload exchanged between the app and the remote host
```

> * Network latency because of the application and remote host physical location. Check if these are in the same network, same geographical region, same country, or the other side of the world.

Reviewer: ```suggestion
* Locations of the application and remote host +
If the remote host is outside of your application's local network, latency is directly proportionate to the remote host response time, regardless of payload size.
```

> * Remote host response time.

Reviewer: ```suggestion
* Remote host response time
```

> The throughput and response time of your application can improve if the remote host is located within the same network boundaries as the application. However, if the remote host is outside of your application's local network, you may experience an increased latency which will be directly proportional to the remote host response time, regardless of the payload size. +

Reviewer: OMIT per suggested rewording:
```suggestion
```

> There are many other factors involved in an application's throughput such as external dependencies, concurrency, flow complexity, deployment characteristics, machine configuration, and so on.

Reviewer: ```suggestion
Other factors that affect application throughput include external dependencies, concurrency, flow complexity, deployment characteristics, and machine configuration.
```
